pkg/cloud/aws/services/ec2/vpc.go

```diff
@@ -140,18 +152,29 @@ func (s *Service) createVPC() (*v1alpha1.VPCSpec, error) {
 	}
 	s.scope.V(2).Info("Created new VPC with cidr", "vpc-id", *out.Vpc.VpcId, "cidr-block", *out.Vpc.CidrBlock)
 
+	// TODO: we should attempt to record the VPC ID as soon as possible by setting s.scope.VPC().ID
+	// however, the logic used for determining managed vs unmanaged VPCs relies on the tags and will
+	// need to be updated to accommodate for the recording of the VPC ID prior to the tagging.
+
 	wReq := &ec2.DescribeVpcsInput{VpcIds: []*string{out.Vpc.VpcId}}
 	if err := s.scope.EC2.WaitUntilVpcAvailable(wReq); err != nil {
 		return nil, errors.Wrapf(err, "failed to wait for vpc %q", *out.Vpc.VpcId)
 	}
 
+	// Apply tags so that we know this is a managed VPC.
 	tagParams := s.getVPCTagParams(*out.Vpc.VpcId)
 	tagApply := &tags.ApplyParams{
 		EC2Client:   s.scope.EC2,
 		BuildParams: tagParams,
 	}
-
-	if err := tags.Apply(tagApply); err != nil {
+	tag := func() (bool, error) {
+		if err := tags.Apply(tagApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), tag, awserrors.VPCNotFound); err != nil {
```

Comment:
@ncdc thoughts on the readability of this if I inlined the `func()` instead of storing it in `tag` above? Same with the `tagApply` and `tagParams`.

Reply:
I've been holding off on suggesting inlining the funcs - I assumed you liked it better that way. If you want to do it, no objections here.

Reply:
I was sticking with the previous convention used with `WaitForWithRetryable`, I much prefer the inlined approach :joy: